core/events/events_block_change.ts

```diff
@@ -31,11 +31,14 @@ export class BlockChange extends BlockBase {
   override type = eventUtils.BLOCK_CHANGE;
   /**
    * The element that changed; one of 'field', 'comment', 'collapsed',
-   * 'disabled', 'invalid', 'inline', or 'mutation'
+   * 'disabled', 'inline', or 'mutation'
    */
   element?: string;
 
-  /** The name of the field that changed, if this is a change to a field. */
+  /**
+   * The name of the field that changed, if this is a change to a field, or
+   * the identifier of the reason why a block can be disabled.
```

Comment:
Done.

Reply:
Are you proposing adding a sixth optional argument to the BlockChange constructor? (And ignore the name argument for "disabled" events?)

Reply:
Yeee I think it's weird because really these should be different events instead of all being kludged onto the same class. Personally, I would prefer not to overload the existing property. But if you think it's better to store things in `name` to avoid extra optional params, I don't have strong opinions about it.

Reply:
After some discussion we chose to add a public method to BlockChange called setDisabledReason() instead of repurposing the name argument of the constructor.